src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -591,20 +617,8 @@ private function walkGroupAst(TreeNode $ast, TrinaryLogic &$isNonEmpty, TrinaryL
 			if ($literalValue !== null) {
 				if (Strings::match($literalValue, '/^\d+$/') === null) {
 					$isNumeric = TrinaryLogic::createNo();
-				}
-
-				if (!$inOptionalQuantification) {
-					$isNonEmpty = TrinaryLogic::createYes();
-				}
-			}
-
-			if ($ast->getValueToken() === 'character_type') {
-				if ($ast->getValueValue() === '\d') {
-					if ($isNumeric->maybe()) {
-						$isNumeric = TrinaryLogic::createYes();
-					}
-				} else {
-					$isNumeric = TrinaryLogic::createNo();
+				} elseif ($isNumeric->maybe()) {
+					$isNumeric = TrinaryLogic::createYes();
```

Comment:
I don't think this else-if can work.

a single `\d` does not turn a expression which exists of more tokens into a numeric-string.

we can assign `$isNumeric = TrinaryLogic::createYes();` only in cases in which we know the whole expression.

Reply:
hmm I tried to find examples which break this logic, but couldn't find any. 

maybe its actually correct ;)

Reply:
hmm I think we need a test for `(.\d)`

Reply:
Ah I see, I missed the by-ref, I guess that's also why $allNumeric = false before the break wasn't strictly needed (but I still think it's safer to include). Fixed in https://github.com/phpstan/phpstan-src/pull/3244/commits/4034c59ea8ca9785cdb28b740f28125861006e98

Reply:
I could also not find a way to trigger a problem the other way, so I just removed it.

Reply:
Ah sorry I'm confusing myself here without full context.. I removed the code below handling \d in character_type as getLiteralValue now handles \d and returns `'0'`, that's why I added the isNumeric here.

I think it works because it can still be set to false/createNo later if some other literal is not a digit.